File: tags/clearsilver-0.9.8/csharp/cstest.cs
using System;
using System.Runtime.InteropServices;

using Clearsilver;

public class CSTest {
   public static unsafe int Main(string[] argv) {
      Hdf h = new Hdf();

      h.setValue("foo.1","1");
      h.setValue("foo.2","2");
      Console.WriteLine("foo.2 = {0}", h.getValue("foo.2","def"));

      CSTContext cs = new CSTContext(h);
//      cs.parseFile("test.cst");
      Console.WriteLine(cs.render());
      
      return 0;
   }
  


}

File: tags/clearsilver-0.9.11/java-jni/CSTest.java

import java.io.*;
import java.util.*;

class CSTest {

    public static void main( String [] args ) throws IOException {
	org.clearsilver.HDF hdf = new org.clearsilver.HDF();

	hdf.setValue("Foo.Bar","10");
        hdf.setValue("Foo.Baz","20");
        System.out.println( hdf.dump() );

	System.out.println( "----" );

	org.clearsilver.CS cs = new org.clearsilver.CS(hdf);
	
	String tmplstr = "Foo.Bar:<?cs var:Foo.Bar ?>\nFoo.Baz:<?cs var:Foo.Baz ?>\n";
	System.out.println(tmplstr);
	System.out.println("----");

	cs.parseStr(tmplstr);
	System.out.println(cs.render());

        // test registered functions
        hdf.setValue("Foo.EscapeTest","abc& 231<>/?");
   
        tmplstr = " <?cs var:url_escape(Foo.EscapeTest) ?> <?cs var:html_escape(Foo.EscapeTest) ?>";

	cs.parseStr(tmplstr);
	System.out.println(cs.render());
    }
};
